src/services/withParams.ts

```diff
@@ -3,21 +3,24 @@ import { ExtractParamName, ExtractWithParamsParamType, ParamEnd, ParamStart } fr
 import { Param } from '@/types/paramTypes'
 import { Identity } from '@/types/utilities'
 import { isRecord } from '@/utilities/guards'
+import { MakeOptional } from '@/utilities/makeOptional'
+
+type AssignableParamsFromString<
+  TValue extends string
+> = TValue extends `${string}${ParamStart}${infer TParam}${ParamEnd}${infer Rest}`
+  ? Record<ExtractParamName<TParam>, Param | undefined> & AssignableParamsFromString<Rest>
+  : {}
 
 type ExtractParamsFromString<
   TValue extends string,
   TParams extends Record<string, Param | undefined> = Record<never, never>
-> = TValue extends `${string}${ParamStart}${infer Param}${ParamEnd}${infer Rest}`
-  ? Record<Param, ExtractWithParamsParamType<Param, TParams>> & ExtractParamsFromString<Rest, TParams>
-  : Record<never, never>
-
-export type ParamsWithParamNameExtracted<TValue extends string> = {
-  [K in keyof ExtractParamsFromString<TValue> as ExtractParamName<K>]?: Param
-}
+> = TValue extends `${string}${ParamStart}${infer TParam}${ParamEnd}${infer Rest}`
+  ? Record<TParam, ExtractWithParamsParamType<TParam, TParams>> & ExtractParamsFromString<Rest, TParams>
+  : {}
```

Comment:
I don't understand why we need both of these. Seems like we only need `AssignableParamsFromString`

Reply:
I'll adjust the names
- `AssignableParamsFromString` is the "input" for `withParams(..., THIS)`
- `ExtractParamsFromString ` is the "ouput" for `withParams(....) => { params: THIS }`